### PR TITLE
Fix:2188 Fixing Generate Config File layout

### LINKED ISSUE #2181: "Generate Config Files" layout is not uniform.
**Actual Behaviour**

The spacing between the instrument and time interval is varing for different instrument.

**Expected Behaviour**

All the instruments should have a similar layout style.

**Screenshots of the issue**

   

**Device**

Asus max pro m1(5.9 inch)

### LINKED ISSUE #2188: Time interval units are not shown to the user.
**Actual Behaviour**

Time interval units are not known to the user when generating a config file.

**Expected Behaviour**

User should be able to know what time interval is selected.

**Screenshots of the issue**

 

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by kartikeysaran**
Fixes #2188
Fixes #2181 

**Changes**: Using ConstraintLayout and Margins to Fix the Generate Config File Layout 
This works on devices of all sizes
**Screenshot/s for the changes**: 

**Checklist**:
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [x] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

### CODE REVIEW

File: app/src/main/res/layout/activity_create_config.xml [hunk 1]
```diff
@@ -32,11 +35,15 @@
...
                     android:layout_height="wrap_content"
+                    android:layout_marginTop="@dimen/create_config_margin1"
                     android:layout_gravity="center"
                     android:text="@string/select_instrument_title" />
 
                 <android.support.v7.widget.AppCompatSpinner
                     android:id="@+id/select_instrument_spinner"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:layout_gravity="center"/>
```
Comment (kartikaysharma01) [inline app/src/main/res/layout/activity_create_config.xml:46]: You have closed the spinner by mistake here

File: app/src/main/res/layout/activity_create_config.xml [hunk 2]
```diff
@@ -32,11 +35,15 @@
...
                     android:layout_width="@dimen/width_zero"
                     android:layout_weight="@dimen/weight_two"
                     android:layout_height="wrap_content"
```
Comment (kartikaysharma01) [inline app/src/main/res/layout/activity_create_config.xml:49]: Please remove the lines off your PR if they are not being used in the code anymore

File: app/src/main/res/layout/activity_create_config.xml [hunk 3]
```diff
@@ -67,12 +75,16 @@
...
                     android:layout_gravity="center"/>
 
                 <android.support.v7.widget.AppCompatSpinner
                     android:id="@+id/interval_unit_spinner"
+                    android:layout_width="wrap_content"
+                    android:layout_gravity="center"
+                    android:layout_height="wrap_content" />
                     android:layout_width="@dimen/width_zero"
                     android:layout_weight="@dimen/weight_one"
                     android:layout_height="wrap_content"
```
Comment (kartikaysharma01) [inline app/src/main/res/layout/activity_create_config.xml:90]: Please remove the lines off your PR if they are not being used in the code anymore

**Comment by kartikaysharma01** [comment]
@kartikeysaran  I am sorry but the solution either seems to be device specific or incomplete. Can you please share the screenshot for Barometer instrument as per your fix

**Comment by kartikaysharma01** [comment]
@kartikeysaran  Also please refer issue #2208 . There is already a PR for that.

**Comment by kartikeysaran** [comment]
> @kartikeysaran I am sorry but the solution either seems to be device specific or incomplete. Can you please share the screenshot for Barometer instrument as per your fix

Please have a look i am using contraint layout and i can share screenshot for each device it works fine with all

**Comment by CloudyPadmal** [comment]
Thanks for the PR @kartikeysaran . But let's go with the card layout.

**Comment by kartikeysaran** [comment]
> Thanks for the PR @kartikeysaran . But let's go with the card layout.

Should i update the pr with Card Layout ?

**Comment by CloudyPadmal** [comment]
#2210 did that

**Pull request closed without merging**